tensorboard/webapp/widgets/data_table/BUILD

```diff
@@ -123,11 +135,37 @@ tf_ng_module(
     ],
 )
 
+tf_ng_module(
+    name = "filter_dialogue",
+    srcs = [
+        "filter_dialogue.ts",
+        "filter_dialogue_module.ts",
+    ],
+    assets = [
+        "filter_dialogue.ng.html",
+        ":filter_dialogue_styles",
+    ],
+    deps = [
+        ":types",
+        "//tensorboard/webapp/angular:expect_angular_material_button",
+        "//tensorboard/webapp/angular:expect_angular_material_checkbox",
```

Comment:
Note that this will need to be added to the legacy component allowlist before being synced internally. cl/547923944

Reply:
The styling looks a little off but I can probably overcome the issue.
![image](https://github.com/tensorflow/tensorboard/assets/78179109/874f3c1a-7e17-43a0-890e-6429f54ef4a0)